packages/babel-types/src/definitions/flow.js

```diff
@@ -142,54 +162,62 @@ defineType("FunctionTypeAnnotation", {
   visitor: ["typeParameters", "params", "rest", "returnType"],
   aliases: ["Flow"],
   fields: {
-    // todo
+    typeParameters: validateOptionalType("TypeParameterDeclaration"),
+    params: validate(arrayOfType("FunctionTypeParam")),
+    rest: validateOptionalType("FunctionTypeParam"),
+    returnType: validateType("Flow"),
   },
 });
 
 defineType("FunctionTypeParam", {
   visitor: ["name", "typeAnnotation"],
   aliases: ["Flow"],
   fields: {
-    // todo
+    name: validateOptionalType("Identifier"),
+    typeAnnotation: validateType("Flow"),
+    optional: validateOptional(assertValueType("boolean")),
   },
 });
 
 defineType("GenericTypeAnnotation", {
   visitor: ["id", "typeParameters"],
   aliases: ["Flow"],
   fields: {
-    // todo
+    id: validateType("Identifier"),
+    typeParameters: validateOptionalType("TypeParameterInstantiation"),
   },
 });
 
 defineType("InferredPredicate", {
   aliases: ["Flow", "FlowPredicate"],
-  fields: {
-    // todo
-  },
 });
 
 defineType("InterfaceExtends", {
   visitor: ["id", "typeParameters"],
   aliases: ["Flow"],
   fields: {
-    // todo
+    id: validateType("Identifier"),
+    typeParameters: validateOptionalType("TypeParameterInstantiation"),
   },
 });
 
 defineType("InterfaceDeclaration", {
   visitor: ["id", "typeParameters", "extends", "body"],
   aliases: ["Flow", "FlowDeclaration", "Statement", "Declaration"],
   fields: {
-    // todo
+    id: validateType("Identifier"),
+    typeParameters: validateOptionalType("TypeParameterDeclaration"),
+    extends: validate(arrayOfType("InterfaceExtends")),
+    mixins: validate(arrayOfType("InterfaceExtends")),
```

Comment:
Just for curiosity, what is this?

Reply:
See https://github.com/facebook/flow/commit/b72d1c61487f9e95ddbe68c4f994d9699b7b71c9